mat64/dense_arithmetic.go

```diff
@@ -251,13 +283,28 @@ func (m *Dense) Inverse(a Matrix) error {
 	}
 	m.reuseAs(a.Dims())
 	aMat, aTrans := untranspose(a)
-	if m != aMat {
-		m.Copy(a)
-	} else if aTrans {
-		tmp := getWorkspace(r, c, false)
-		tmp.Copy(a)
-		m.Copy(tmp)
-		putWorkspace(tmp)
+	switch rm := aMat.(type) {
+	case RawMatrixer:
+		if m != aMat || aTrans {
+			if m == aMat || m.checkOverlap(rm.RawMatrix()) {
+				tmp := getWorkspace(r, c, false)
+				tmp.Copy(a)
+				m.Copy(tmp)
+				putWorkspace(tmp)
+				break
+			}
+			m.Copy(a)
+		}
+	default:
+		if m != aMat {
```

Comment:
If aMat is not a RawMatrixer, then it is impossible for m to equal aMat, isn't it? Thus it should just be m.Copy(a)?